nixos/modules/hardware/video/hidpi.nix

```diff
@@ -11,6 +11,36 @@ with lib;
     console.earlySetup = mkDefault true;
     boot.loader.systemd-boot.consoleMode = mkDefault "1";
 
+    # Use a scalable font for the X11 "core" cursor
+    #
+    # If not running a fancy desktop environment, the cursor is likely set to
+    # the default `cursor.pcf` bitmap font. This is 17px wide, so it's very
+    # small and almost invisible on 4K displays.
+    nixpkgs.overlays = mkIf (config.services.xserver.dpi != null)
```

Comment:
I don't think we should use an overlay, because overlays cause unintended consequences when they affect other use cases of a package.
You could override `fontcursormisc` at its use site in `nixos/modules/config/fonts/fonts.nix` instead.

Reply:
> when they affect other use cases of a package

I'm not sure what could be other use cases of this package. It literally does one thing: provide the default cursor used by xorg-server.

> You could override fontcursormisc at its use site in nixos/modules/config/fonts/fonts.nix instead.

I would like to keep this tied to `hardware.video.hidpi.enable`, because the bitmap version looks better than the scaled one on a normal 96dpi dispay. I don't think it's possible to do this without an overlay.

Reply:
Xquartz and xdummy also use it. It might not cause an issue because those are used independently of NixOS, presumably. Nonetheless, let's keep the code simple.

> I would like to keep this tied to `hardware.video.hidpi.enable` [...] I don't think it's possible to do this without an overlay.

It's possible. You can change `pkgs.xorg.fontcursormisc` in `fonts.nix` to include the override and make it optional based on `config.hardware.video.hidpi.enable`.

You can read from options declared by other modules. It doesn't happen that frequently because usually it isn't necessary, but it's a valid thing to do here as `hidpi.enable` is a cross-cutting concern. Alternatively, you could introduce a new option in the `fonts.nix` module, to be configured by `hidpi.nix`.

Reply:
> It's possible. You can change pkgs.xorg.fontcursormisc in fonts.nix

Ah, you mean this:

> 
>    fonts.fonts = mkIf config.fonts.enableDefaultFonts
>      ([
>        pkgs.dejavu_fonts
>        pkgs.freefont_ttf
>        pkgs.gyre-fonts # TrueType substitutes for standard PostScript fonts
>        pkgs.liberation_ttf
>        pkgs.xorg.fontmiscmisc
>        **pkgs.xorg.fontcursormisc**
>        pkgs.unifont
>        pkgs.noto-fonts-emoji
>      ] ++ lib.optionals (config.nixpkgs.config.allowUnfree or false) [
>        # these are unfree, and will make usage with xserver fail
>        pkgs.xorg.fontbhlucidatypewriter100dpi
>        pkgs.xorg.fontbhlucidatypewriter75dpi
>        pkgs.xorg.fontbh100dpi
>      ]);

Ok, this work but I'll have to move the `enableDefaultFonts` gate, somehow.
What happens is that when fontcursormisc is not installed, the xorg-server somehow manages to find it anyway and it would ignore the patched package inserted here.
I should put it behind a `services.xserver.enable` gate, or maybe unconditionally install it.